index.js

```diff
@@ -18,6 +20,11 @@
     return result;
   };
 
+  function generateRegExp(props) {
```

Comment:
I suggest to avoid code building in such cases. It's harder to read and debug

```suggestion
  const pattern = "(export default {\s*)(name ?:|extends ?:|watch ?:|methods ?:|props ?:|model ?:|computed ?:|components ?:|mixins ?:|filters ?:|data ?[:(](.*){)"
```